Twitter embedded timeline: Add "verified" account example

EricDunsworth opened the pull request:
Timeline widgets currently work differently between standard and "verified" Twitter accounts:
* Standard accounts: Shows a "Nothing to see here - yet" placeholder message
* "Verified" accounts: Shows top 100 most-liked tweets (ignores ``data-tweet-limit`` attribute)

This adds a second timeline for @Canada ("verified" government account) to demonstrate the difference between both "types" of account timelines.

Other changes:
* Use a ``data-height="fb-page"`` attribute for the "verified" account example (via #9691)
* Adjust nearby content to be more presentable:
  * Revise example heading (change to plural, update timeline's official name, use sentence case)
  * Revamp Twitter docs link into an expanded section
* Update French translations:
  * Translate main page content
  * Update outdated page description translation

duboisp commented on src/plugins/twitter/twitter-fr.hbs:
There is not a lot of content on this page, let's translate it in French.

If you can't, please let us know and we will propose a French translation.

EricDunsworth replied:
I'll try to take a shot at it :).

EricDunsworth replied:
Updated :)!

duboisp commented on src/plugins/twitter/twitter-fr.hbs:
```diff
@@ -20,18 +20,39 @@
+		<section class="col-md-6">
+			<h3>Normal account</h3>
+			<div class="wb-twitter">
+				<a class="twitter-timeline" href="https://twitter.com/WebExpToolkit" data-tweet-limit="3">Tweets by @WebExpToolkit</a>
```
Indicate to produce a French embedded timeline. We can configure this via `data-lang="fr"`.

```suggestion
				<a class="twitter-timeline" href="https://twitter.com/WebExpToolkit" data-tweet-limit="3" data-lang="fr">Tweets by @WebExpToolkit</a>
```

EricDunsworth replied:
I think I'll take a different approach for this.

Instead of hardcoding ``data-lang``, I'll modify the plugin to automatically inject ``data-lang`` based on the widget's "contextual" language (i.e. parent element or page language). That'll spare implementers from needing to hardcode widget languages.

Btw once the widget loads, the language it displays in doesn't influence tweet or account link destinations (i.e. it won't add a URL param to change Twitter's UI lang to whatever language the widget's UI was using).

EricDunsworth replied:
Marking this as a resolved due to https://github.com/wet-boew/wet-boew/pull/9691#discussion_r1385258973.

duboisp commented on src/plugins/twitter/twitter-fr.hbs:
```diff
@@ -20,18 +20,39 @@
+			<section>
+				<h4>Code</h4>
+				<pre><code>&lt;div class=&quot;wb-twitter&quot;&gt;
+	&lt;a class=&quot;twitter-timeline&quot; href=&quot;https://twitter.com/WebExpToolkit&quot; data-tweet-limit=&quot;3&quot; &gt;Tweets by @WebExpToolkit&lt;/a&gt;
```
```suggestion
	&lt;a class=&quot;twitter-timeline&quot; href=&quot;https://twitter.com/WebExpToolkit&quot; data-tweet-limit=&quot;3 data-lang="fr"&quot; &gt;Tweets by @WebExpToolkit&lt;/a&gt;
```

EricDunsworth replied:
Skipping/marking as resolved due to https://github.com/wet-boew/wet-boew/pull/9675#discussion_r1375055677.

duboisp commented on src/plugins/twitter/twitter-fr.hbs:
```diff
@@ -20,18 +20,39 @@
+		<section class="col-md-6">
+			<h3>"Verified" account</h3>
+			<div class="wb-twitter">
+				<a class="twitter-timeline" href="https://twitter.com/Canada" data-tweet-limit="3">Tweets by @Canada</a>
```
```suggestion
				<a class="twitter-timeline" href="https://twitter.com/Canada" data-tweet-limit="3" data-lang="fr">Tweets by @Canada</a>
```

EricDunsworth replied:
Skipping/marking as resolved due to https://github.com/wet-boew/wet-boew/pull/9675#discussion_r1375055677.

duboisp commented on src/plugins/twitter/twitter-fr.hbs:
```diff
@@ -20,18 +20,39 @@
+			<section>
+				<h4>Code</h4>
+				<pre><code>&lt;div class=&quot;wb-twitter&quot;&gt;
+	&lt;a class=&quot;twitter-timeline&quot; href=&quot;https://twitter.com/Canada&quot; data-tweet-limit=&quot;3&quot; &gt;Tweets by @Canada&lt;/a&gt;
```
```suggestion
	&lt;a class=&quot;twitter-timeline&quot; href=&quot;https://twitter.com/Canada&quot; data-tweet-limit=&quot;3 data-lang="fr"&quot; &gt;Tweets by @Canada&lt;/a&gt;
```

EricDunsworth replied:
Skipping/marking as resolved due to https://github.com/wet-boew/wet-boew/pull/9675#discussion_r1375055677.

duboisp reviewed (changes requested):
Regarding the "data-limit" which are not longer applied, that could impact the design of some page.

Would you like to fix that issue with through this content PR or via a new PR?

EricDunsworth commented on src/plugins/twitter/twitter-fr.hbs:
```diff
@@ -17,21 +17,39 @@
+		<section class="col-md-6">
+			<h3>Compte «&nbsp;certifié&nbsp;»</h3>
+			<div class="wb-twitter">
+				<a class="twitter-timeline" href="https://twitter.com/Canada" data-tweet-limit="2">Tweets de @Canada</a>
```
I would've preferred using [@AuCanada](https://twitter.com/AuCanada) (i.e. official French version of [@Canada](https://twitter.com/Canada)) for this example, but... Twitter doesn't currently classify it as a "verified" government account.

[@AuCanada](https://twitter.com/AuCanada)'s widget works the same way as a standard account and trying to view the account's profile shows a login wall. So it can't be used as a "verified" example at this time :(.

EricDunsworth committed:
Twitter embedded timeline: Add "verified" account example

Timeline widgets currently work differently between standard and "verified" Twitter accounts:
* Standard accounts: Shows a "Nothing to see here - yet" placeholder message
* "Verified" accounts: Shows top 100 most-liked tweets (ignores data-tweet-limit attribute)

This adds a second timeline for @Canada ("verified" government account) to demonstrate the difference between both "types" of account timelines.

Other changes:
* Use a data-height="fb-page" attribute for the "verified" account example (via #9691)
* Adjust nearby content to be more presentable:
  * Revise example heading (change to plural, update timeline's official name, use sentence case)
  * Revamp Twitter docs link into an expanded section
* Update French translations:
  * Translate main page content
  * Update outdated page description translation

EricDunsworth commented:
This should be a good to re-review now. I'd recommend merging it after #9691.

Btw if this PR is tested on its own before #9691 has been merged, the new example's widget height will be insanely tall.

The best way of testing would probably be as follows:
1. Clone my WET fork locally (``https://github.com/EricDunsworth/wet-boew.git``).
2. Switch to this PR's branch (``EricDunsworth:v4.0-twitter-verified-example``).
3. Run ``grunt``.
4. (Optional) Check the Twitter demo pages (verified example will be incredibly tall).
5. Switch to #9691's branch (``EricDunsworth:v4.0-twitter-data-presets``).
6. Run ``grunt js``.
7. (Optional) Run ``grunt docs`` if you also want to review that PR's updated docs.
8. Check the Twitter demo pages (verified example's height will be 500px).

Good luck :)!

duboisp commented:
I suggest to merge this one with the other PR because it do leverage new feature added by #9691

EricDunsworth commented:
Superseded by #9704.
